core/src/main/java/org/elasticsearch/index/query/NotQueryParser.java

```diff
@@ -30,7 +30,7 @@
  */
 public class NotQueryParser extends BaseQueryParser<NotQueryBuilder> {
 
-    private static final ParseField QUERY_FIELD = new ParseField("filter", "query");
+    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
```

Comment:
nasty... we have to backport this to 2.x and 2.0, and master if not filter is still there.